core/src/test/scala/unit/kafka/server/FetchRequestTest.scala

```diff
@@ -227,7 +228,7 @@ class FetchRequestTest extends BaseRequestTest {
       val suffix = s"$tp-$messageIndex"
       new ProducerRecord(tp.topic, tp.partition, s"key $suffix", s"value $suffix")
     }
-    records.map(producer.send).foreach(_.get)
+    records.map(producer.send(_).get)
```

Comment:
Right. A lot of the testing around the max message size is obviously sensitive to how the messages are batched. If possible, I'd like to save refactoring of these test cases for a follow-up.